integration/container_event_test.go

```diff
@@ -0,0 +1,133 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/containerd/containerd/integration/images"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+func TestContainerEvents(t *testing.T) {
+	ctx := context.Background()
+	defer ctx.Done()
+
+	t.Log("Set up container events streaming client")
+	containerEventsStreamingClient, err := runtimeService.GetContainerEvents(ctx, &runtime.GetEventsRequest{})
+	assert.NoError(t, err)
+	containerEventsChan := make(chan *runtime.ContainerEventResponse)
+
+	go listenToEventChannel(ctx, t, containerEventsChan, containerEventsStreamingClient)
+	// Sleep to make sure the event channel is set up, and then drain all events
+	// emitted by previous tests.
+	time.Sleep(2 * time.Second)
+	drainContainerEventsChan(t, containerEventsChan)
+
+	t.Logf("Create a pod config and run sandbox container")
+	sandboxName := "container_events_sandbox"
+	sb, sbConfig := PodSandboxConfigWithCleanup(t, sandboxName, "container_events")
+
+	assert.NoError(t, checkContainerEventResponse(t, containerEventsChan, runtime.ContainerEventType_CONTAINER_CREATED_EVENT, sandboxName, ""))
+	assert.NoError(t, checkContainerEventResponse(t, containerEventsChan, runtime.ContainerEventType_CONTAINER_STARTED_EVENT, sandboxName, ""))
```

Comment:
Looks like a duplicate assertion

```suggestion
	assert.NoError(t, checkContainerEventResponse(t, containerEventsChan, runtime.ContainerEventType_CONTAINER_CREATED_EVENT, sandboxName, ""))
```

Reply:
no the event type are different. I am asserting CONTAINER_CREATED_EVENT and then STARED.

Reply:
Thanks, I missed that.